### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by robskillington**

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://github.com/m3db/m3/blob/master/CONTRIBUTING.md and developer notes https://github.com/m3db/m3/blob/master/DEVELOPER.md
+2. Please label this pull request according to what type of issue you are addressing.
```
Comment (benraskin92) [inline .github/pull_request_template.md]: also maybe add: e.g. if this is a pr for the query service add `[query]` to the title of your pr

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,23 +1,15 @@
 Contributing
 ============
 
-We'd love your help making M3DB great!
+We'd love your help making M3DB better!
```
Comment (benraskin92) [inline CONTRIBUTING.md]: haha love it

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -29,4 +21,8 @@ pull request is most likely to be accepted if it:
   review comments](https://github.com/golang/go/wiki/CodeReviewComments).
 * Has a [good commit
   message](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- 
\ No newline at end of file
+
+## Developing
+
+See the (developer notes)[https://github.com/m3db/m3/blob/master/DEVELOPER.md]
```
Comment (benraskin92) [inline CONTRIBUTING.md]: `(developer notes)[https://github.com/m3db/m3/blob/master/DEVELOPER.md]` => `[developer notes](https://github.com/m3db/m3/blob/master/DEVELOPER.md)`

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+3. Ensure you have added or ran the appropriate tests for your PR: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#testing-changes
+4. Follow the instructions for writing a changelog note: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#adding-a-changelog
+-->
+
+**What this PR does / why we need it**:
+
+**Which issue(s) this PR fixes**:
+<!--
+*Automatically closes linked issue when PR is merged.
```
Comment (richardartoul) [inline .github/pull_request_template.md]: does this work for us? Not sure if this is some custom kube thing

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
+-->
+Fixes #
+
+**Special notes for your reviewer**:
+
+**Does this PR introduce a user-facing change?**:
```
Comment (richardartoul) [inline .github/pull_request_template.md]: Some concept of whether its backwards compat / a breaking change

File: CONTRIBUTING.md (hunk 2 above)
Comment (richardartoul) [reply CONTRIBUTING.md]: nice

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://github.com/m3db/m3/blob/master/CONTRIBUTING.md and developer notes https://github.com/m3db/m3/blob/master/DEVELOPER.md
+2. Please label this pull request according to what type of issue you are addressing.
+3. Ensure you have added or ran the appropriate tests for your PR: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#testing-changes
+4. Follow the instructions for writing a changelog note: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#adding-a-changelog
+-->
+
+**What this PR does / why we need it**:
+
+**Which issue(s) this PR fixes**:
```
Comment (benraskin92) [inline .github/pull_request_template.md]: I'd say just remove this section and combine it with the `What this PR does / why we need it` section. You can add a comment saying that you should list out the issues that this pr resolves.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://github.com/m3db/m3/blob/master/CONTRIBUTING.md and developer notes https://github.com/m3db/m3/blob/master/DEVELOPER.md
+2. Please label this pull request according to what type of issue you are addressing.
+3. Ensure you have added or ran the appropriate tests for your PR: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#testing-changes
+4. Follow the instructions for writing a changelog note: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#adding-a-changelog
+-->
+
+**What this PR does / why we need it**:
+
```
Comment (benraskin92) [inline .github/pull_request_template.md]: nit, but can we add this to each section? (like you did for the release-note and documentation-note)
\```release-note
\```

File: .github/pull_request_template.md (hunk 4 above)
Comment (schallert) [reply .github/pull_request_template.md]: https://help.github.com/en/articles/closing-issues-using-keywords

File: CONTRIBUTING.md (hunk 3 above)
Comment (robskillington) [reply CONTRIBUTING.md]: ta

File: .github/pull_request_template.md (hunk 1 above)
Comment (robskillington) [reply .github/pull_request_template.md]: Sure, it will be hard to enforce... but I can add it.

File: .github/pull_request_template.md (hunk 4 above)
Comment (robskillington) [reply .github/pull_request_template.md]: It does yes.

File: .github/pull_request_template.md (hunk 5 above)
Comment (robskillington) [reply .github/pull_request_template.md]: That's actually implied, but I'll update it nonetheless to "Does this PR introduce a user-facing and/or backwards incompatible change?".

File: .github/pull_request_template.md (hunk 6 above)
Comment (robskillington) [reply .github/pull_request_template.md]: Sure, combined and kept the issue fixing text.

File: .github/pull_request_template.md (hunk 7 above)
Comment (robskillington) [reply .github/pull_request_template.md]: Hm no, those are used to specify sub snippets that may be automatically added to the changelog at a later time.

The text of the description will not be automatically appended anywhere except git history.

**Pull request merged**